src/Lexer.php

```diff
@@ -61,6 +63,15 @@ public function __construct(Environment $env, array $options = [])
             'whitespace_line_chars' => ' \t\0\x0B',
             'interpolation' => ['#{', '}'],
         ], $options);
+    }
+
+    private function initialize()
+    {
+        if ($this->isInitialized) {
```

Comment:
instead of adding a new property, you could do `if (isset($this->regexes))`

Reply:
but `if ($this->isInitialized)` is self-explanatory and `if (isset($this->regexes))` is like using a side-effect of having or not some regular expressions. It looks much more confusing to me.

Reply:
I agree with Javier.